src/toil_scripts/lib/files.py

```diff
@@ -49,6 +54,58 @@ def copy_file_job(job, name, file_id, output_dir):
     copy_files([fpath], output_dir)
 
 
+def copy_to(filename, work_dir, output_dir):
+    """
+    Copies file or directory to output directory
+
+    :param str filename: basename of file path
+    :param str work_dir: working directory path
+    :param str output_dir: the output directory path
+    :return path to copied file or directory
+    :rtype: str
+    """
+    origin = os.path.abspath(os.path.join(work_dir, filename))
+    dest = os.path.join(output_dir, filename)
+
+    # This method will copy a file or directory.
+    # http://stackoverflow.com/questions/1994488/copy-file-or-directory-in-python
+    try:
+        shutil.copytree(origin, dest)
+    except OSError as e:
+        if e.errno == errno.ENOTDIR:
+            mkdir_p(output_dir)
+            shutil.copy(origin, dest)
+        else:
+            raise e
+    assert os.path.exists(dest)
+    return dest
+
+
+def upload_or_move_job(job, filename, file_id, output_dir, s3_key_path=None):
```

Comment:
I can respect wanting DRY code,  but you're adding 24 lines of code to save ~8 lines?  I don't know what the best route here is.